### PR TITLE
[PUBDEV-7064] Fix NPE in GBM when sample rate < 1

### PR DESCRIPTION
**Pull request by honzasterba**

### CODE REVIEW

File: h2o-algos/src/main/java/hex/tree/gbm/GBM.java [hunk 1]
```diff
@@ -954,10 +954,6 @@ public HuberLeafMath(FrameMap frameMap, double huberDelta, Vec strata) {
 
     @Override
     public void map(Chunk[] cs) {
-      if (_strataMin < 0 || _strataMax < 0) {
-        Log.warn("No Huber math can be done since there's no strata.");
-        return;
-      }
```
Comment (maurever) [inline h2o-algos/src/main/java/hex/tree/gbm/GBM.java]: What is the reason to remove these lines of code? I am just curious. :)

File: h2o-algos/src/main/java/hex/tree/gbm/GBM.java [hunk 2]
```diff
@@ -954,10 +954,6 @@ public HuberLeafMath(FrameMap frameMap, double huberDelta, Vec strata) {
...
       final int nstrata = _strataMax - _strataMin + 1;
       Log.info("Computing Huber math for (up to) " + nstrata + " different strata.");
       _huberGamma = new double[nstrata];
```
Comment (maurever) [inline h2o-algos/src/main/java/hex/tree/gbm/GBM.java]: I am surprised this line of code does not solve the problem with the initialization of `_huberGamma`...  What is the case this line does not initialize the _huberGamma? The zero-length chunk?

File: h2o-algos/src/main/java/hex/tree/gbm/GBM.java (hunk 1 above)
Comment (honzasterba) [reply h2o-algos/src/main/java/hex/tree/gbm/GBM.java]: its does not make any sense, the strata can be negative, even when it is negative its still useful

File: h2o-algos/src/main/java/hex/tree/gbm/GBM.java (hunk 2 above)
Comment (honzasterba) [reply h2o-algos/src/main/java/hex/tree/gbm/GBM.java]: you are partly right, let me improve this

Comment (maurever) [reply h2o-algos/src/main/java/hex/tree/gbm/GBM.java]: Ok, looks more understandable. :+1:

**Pull request merged**